shared/js/background/events.js

```diff
@@ -274,6 +274,18 @@ browser.tabs.onActivated.addListener(() => {
     browserWrapper.notifyPopup({ closePopup: true })
 })
 
+// Include the performanceWarning flag in breakage reports.
+if (browser.runtime.onPerformanceWarning) {
+    browser.runtime.onPerformanceWarning.addListener(({ tabId }) => {
+        if (tabId && tabId > 0) {
+            const tab = tabManager.get({ tabId })
+            if (tab) {
+                tab.performanceWarning = true
+            }
+        }
+    })
+}
```

Comment:
I'd prefer if this was in a separate file - I'm trying to move us away from having everything registered in `events.js`. Instead we could:
- Move this to a 'component', that runs this code in the constructor.
- Instantiate this component in background.js only for Firefox builds.

This has the advantage that it makes this code much easier to test. I see you had to add a load of stubs to handle importing `events.js` in a unit-test. Using the component architecture would remove the need for that, and you could also pass in a mock `tabManager` instance.